qcodes/instrument_drivers/Spectrum/M4i.py

```diff
@@ -777,22 +786,41 @@ def start_acquisition(self, mV_range, memsize, posttrigger_size=None, verbose=0)
 
         return {'memsize': memsize, 'numch': numch, 'mV_range': mV_range}
 
-    def _transfer_buffer_numpy(self, memsize, numch):
-        """ Transfer buffer to numpy array """
+    def _transfer_buffer_numpy(self, memsize, numch, bytes_per_sample=2):
+        """ Transfer buffer to numpy array
+
+        Args:
+            memsize (int): number of samples to transfer
+            numch (int): number of channels
+            bytes_per_sample (int): specifies the datatype. 2 for int16, 4 for int32
+        Returns:
+            array: transfered data
+
+        """
         # setup software buffer
-        buffer_size = ct.c_int16 * memsize * numch
+        if bytes_per_sample == 2:
+            buffer_size = ct.c_int16 * memsize * numch
+        elif bytes_per_sample == 4:
+            buffer_size = ct.c_int32 * memsize * numch
+        else:
+            raise ValueError('bytes_per_sample should be 2 or 4')
         data_buffer = (buffer_size)()

```

Comment:
`buffer_size` is very bad name for the variable. It's not the size, it's a ctype object for a data_buffer of a correct size.